src/auth/auth-manager.ts

```diff
@@ -1,62 +1,86 @@
-  return window.localStorage.getItem(typeToken);
-};
+  const saveUserDataLocalStorage: Function = (token: AuthenticationResponseType) => {
+    window.localStorage.setItem(TokenType.idToken, token?.idToken);
+    window.localStorage.setItem(TokenType.accessToken, token?.accessToken);
+  };
 
-export const isValidAccessToken: Function = () => {
-  const token = getToken(TokenType.access);
-  if (!token) {
-    return false;
-  }
-  const accessTokenDecode: AccessTokenDecodeType = jwtDecode(token);
-  const expDate = new Date(accessTokenDecode?.exp * 1000);
-  return expDate > new Date();
-};
+  const getAccessToken: Function = () => {
+    return window.localStorage.getItem(TokenType.accessToken);
+  };
 
-export const refreshUserSession: Function = async () => {
-  try {
-    const refreshToken: string | null = getToken(TokenType.refresh);
-    if (refreshToken) {
-      const res: AuthenticationResponseType = await refreshSession(
-        refreshToken
-      );
-      saveUserDataLocalStorage({ ...res });
-      return true;
+  const getIdToken: Function = () => {
+    return window.localStorage.getItem(TokenType.idToken);
+  };
+
+  const loginUser = async () => {
+    const result = await msalInst?.loginPopup();
+    saveUserDataLocalStorage(result);
+  };
+
+  const logoutUser: Function = async () => {
+    const currentAccount: AccountInfo | undefined =
+      msalInst?.getAllAccounts()[0];
+    await msalInst?.logoutPopup({ account: currentAccount });
+
+    Object.keys(TokenType).forEach((token) =>
+      window.localStorage.removeItem(token)
+    );
+  };
+
+  const isValidToken: Function = () => {
+    const idToken: any = getIdToken() || '';
+    if (!idToken) {
+      return false;
     }
-    return false;
-  } catch (error: any) {
-    alert(error.message);
-    return false;
-  }
-};
+    const idTokenDecode: tokenDecodeType = jwtDecode(idToken);
+    const expDate = new Date(idTokenDecode?.exp * 1000);
+    return expDate > new Date();
+  };
+
+  const refreshUserSession: Function = async () => {
+    const currentAccount: AccountInfo | undefined =
+      await msalInst?.getAllAccounts()[0];
+    const request = {
+      scopes: ['Mail.Read'],
+      account: currentAccount,
+      forceRefresh: true,
+    };
+    const refreshSession = await msalInst
+      ?.acquireTokenSilent(request)
+      .catch(async (error: any) => {
+        if (error instanceof InteractionRequiredAuthError) {
+          await msalInst?.acquireTokenRedirect(request);
+        }
+      });
+    saveUserDataLocalStorage(refreshSession);
+    return true;
+  };
 
-export const loginUser = async (user: RequestUser) => {
-  const res: AuthenticationResponseType = await login({ ...user });
-  saveUserDataLocalStorage(res);
+  return {
+    init: async () => {
+      const msalInstance = new PublicClientApplication(msalConfig);
+      msalInst = msalInstance;
+      await msalInst.initialize();
+    },
```

Comment:
Let's define this function above just like the others. Then you could shorten the return like this:
return { init, logout: logoutUser, login: loginUser, ....}.